### PR TITLE
genpolicy-msft: add support for volumeDevices

### PR DESCRIPTION
**Pull request by burgerdev**
This is essentially kata-containers/kata-containers#9706, but adopted to Microsoft's fork.

Since the patch touches `rules.rego`, we can't take the one from the URL anymore, so I changed the rules to small derivations that extract the patched `rules.rego` and `genpolicy-settings.json` files, respectively.

### CODE REVIEW

File: packages/by-name/genpolicy-msft/package.nix [hunk 1]
```diff
@@ -25,6 +26,17 @@ rustPlatform.buildRustPackage rec {
     hash = "sha256-W36RJFf0MVRIBV4ahpv6pqdAwgRYrlqmu4Y/8qiILS8=";
   };
 
+  patches = [
+    # TODO(burgerdev): drop this patch after upstream applied it
```
Comment (katexochen) [inline packages/by-name/genpolicy-msft/package.nix:30]: Can you link the upsteam PR?

File: packages/by-name/genpolicy-msft/package.nix [hunk 2]
```diff
@@ -3,8 +3,9 @@
 
 { lib
 , fetchFromGitHub
-, fetchurl
+, fetchpatch
 , applyPatches
+, pkgs
```
Comment (katexochen) [inline packages/by-name/genpolicy-msft/package.nix:8]: we never pass `pkgs` to a package expression. The idea of `callPackage` is to only pass the args needed.

File: packages/by-name/genpolicy-msft/package.nix [hunk 3]
```diff
@@ -49,14 +61,16 @@ rustPlatform.buildRustPackage rec {
   '';
 
   passthru = rec {
-    settings = fetchurl {
+    settings = pkgs.stdenv.mkDerivation {
```
Comment (katexochen) [inline packages/by-name/genpolicy-msft/package.nix:64]:
```suggestion
    settings = stdenvNoCC.mkDerivation {
```

**Review by katexochen — CHANGES_REQUESTED** [review]

File: packages/by-name/genpolicy-msft/package.nix (hunk 2 above)
Comment (burgerdev) [reply packages/by-name/genpolicy-msft/package.nix]: Thanks, changed.

File: packages/by-name/genpolicy-msft/package.nix (hunk 1 above)
Comment (burgerdev) [reply packages/by-name/genpolicy-msft/package.nix]: Done

**Review by katexochen — APPROVED** [review]

**Pull request merged**